### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by SaneMyburg**
### In this PR: ###

 -  #### Created 3 classes: ####
   1. Person
   2. Student 
   3. Teacher

### CODE REVIEW

**Review by BregornOriginal — COMMENTED** [review]
Hi @SaneMyburg  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

***You have submitted a project, that doesn´t contain a Pull Request description. Kindly add a description to your Pull Request to be able for review***

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by FrederickMih — COMMENTED** [review]
## Hi :raising_hand:  @SaneMyburg, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Since the app is not yet deployed, let's get rid of the **Live Demo** section of the README file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(id, age, parent_permission: true, name: 'Unknown')
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] the ID should be created in the initialize method it should not be passed down, have a look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for more details.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(id, age, classroom, parent_permission: true, name: 'Unknown')
+    super(id, age, parent_permission, name)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [ ] the id should be created in the initialize method of the person class it should not be set by the user, have a look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for more details.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(id, age, specialization, parent_permission: true, name: 'Unknown')
+    super(id, age, parent_permission, name)
+    @specialization = specialization
+  end
```
Comment (mikethreels) [inline teacher.rb]: - [ ] the id should be created in the initialize method of the person class it should not be set by the user, have a look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for more details.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Sane, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting up the classes.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AlphaNtihinduka — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 
## Hi @SaneMyburg 🙋‍♂️, 

Your project is complete!  There is nothing else to say other than... Great job 👏 

approved-accepted

Congratulations! 🎉 
TO HIGHLIGHT:

- Each class has been saved in a separate file ✔️ 
- Correct use of classes ✔️ 

Now you can merge this branch and move on :1st_place_medal: 


## Optional suggestions
N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**